File: java/371.java
class Solution {
    public int getSum(int a, int b) {
        if(a == 0){
            return b;
        }
        if(b == 0){
            return a;
        }
        int x = 0xFFFFFFFF,y = 0x7FFFFFFF;
        while(b != 0){
            int carry = a & b;
            a ^= b;
            b = carry<<1;
        }
        return a ;
    }
}

File: go/371.go
import "fmt"

func getSum(a int, b int) int {
	if a == 0 {
		return b
	}
	if b == 0 {
		return a
	}
	for b != 0 {
		carry := a & b
		a ^= b
		b = carry << 1
		fmt.Println(a, b)
	}
	return a
}